Feature/phpstan

jeroenjans opened the pull request:
This PR adds support for [PHPStan](https://github.com/phpstan/phpstan) using the CheckStyleParser.

uhafner commented on src/main/java/io/jenkins/plugins/analysis/warnings/PhpStan.java:
```diff
@@ -0,0 +1,63 @@
+public class PhpStan extends ReportScanningTool {
+    private static final long serialVersionUID = 2699509705079011738L;
+
+    static final String ID = "phpstan";
```
private

uhafner replied:
(I changed that in all other parsers as well:-)

jeroenjans replied:
I've updated the visibility to private.
I've also rebased on master to resolve the merge conflict in the SUPPORTED-FORMATS.md file

jeroenjans commented:
Regarding the image I haven't been able to find information on the license of the icon.
I'll see if I can get some information from the developers on this.

jeroenjans commented:
Alright, I was able to check it with the developers and I was told that we can use the logo for this purpose.
I've added it in the last commit and updated the supported documents file.

uhafner commented:
Nice, thanks! Can you please add a line or reference to the icon license (or to the discussion) in the LICENSE.txt file in the icons folder?

uhafner commented:
And one more thing, can you add a small test case to [ParsersITest](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java#L70) that shows that your parser correctly finds *n* warnings. No need to check for individual properties, since this is done in the check style parser test.

uhafner commented:
Thank you very much!
